nixos/modules/config/sysctl.nix

```diff
@@ -59,5 +59,18 @@ in
     # Disable YAMA by default to allow easy debugging.
     boot.kernel.sysctl."kernel.yama.ptrace_scope" = mkDefault 0;
 
+    # Dynamic since kernel v5.12
+    # https://github.com/torvalds/linux/commit/ac7b79fd190b02e7151bc7d2b9da692f537657f3
+    boot.kernel.sysctl."fs.inotify.max_user_instances" =
+      if (config.boot.kernelPackages.kernel.kernelAtLeast "5.12")
+      then mkDefault 2147483647 # INT_MAX, dynamically limited based on available memory
```

Comment:
Can we make `boot.kernel.INT_MAX` a variable?

Reply:
I guess, it would be more like in the `stdenv`, though or `lib` or even `builtin`. I can make a quick follow-up PR once this is merged.